### ISSUE TITLE
PowerShell on Target Machines task: preference variable "ErrorActionPreference" or common parameter is set to Stop

### ISSUE THREAD

**Issue by rafaelzm2000** [Microsoft/vsts-tasks#3991]
I am using PowerShell on Target Machines task to run a couple of scripts during the release process in Azure. However, the PowerShell on Target Machines task is failing to run properly. As I mentioned I am running two scripts, the first script creates a DB and PowerShell on Target Machines task runs without issue. However, the PowerShell task that runs a price node loader, which fills the DB, fail to run and returns the following message: 

##[error]System.Management.Automation.RuntimeException: The running command stopped because the preference variable "ErrorActionPreference" or common parameter is set to Stop: System.Management.Automation.RemoteException ---> System.Management.Automation.RuntimeException: The running command stopped because the preference variable "ErrorActionPreference" or common parameter is set to Stop: System.Management.Automation.RemoteException

I tried consolidating the scripts into one but that fail too. I tried adding a pause between running the scripts no luck either.

Please keep in mind if I run these scripts locally on the target machine they work perfectly. Any help will be much appreciated.
complete log below.

-------------------------------------------------------------------------------------------------------------------

2017-04-10T19:09:25.9867536Z ##[section]Starting: Run PowerShell on xyz.eastus.cloudapp.azure.com
2017-04-10T19:09:25.9867536Z ==============================================================================
2017-04-10T19:09:25.9867536Z Task         : PowerShell on Target Machines
2017-04-10T19:09:25.9867536Z Description  : Execute PowerShell scripts on remote machine(s)
2017-04-10T19:09:25.9867536Z Version      : 1.0.47
2017-04-10T19:09:25.9867536Z Author       : Microsoft Corporation
2017-04-10T19:09:25.9867536Z Help         : [More Information](https://go.microsoft.com/fwlink/?linkid=627414)
2017-04-10T19:09:25.9867536Z ==============================================================================
2017-04-10T19:09:25.9877753Z Preparing task execution handler.
2017-04-10T19:09:26.3047322Z Executing the powershell script: d:\a\_tasks\PowerShellOnTargetMachines_3b5693d4-5777-4fee-862a-bd2b7a374c68\1.0.47\PowerShellOnTargetMachines.ps1
2017-04-10T19:09:27.9548780Z Deployment started for machine: 'xyz.eastus.cloudapp.azure.com:5986'
2017-04-10T19:09:34.9950209Z 	 Deployment started on target machine...
2017-04-10T19:09:34.9950209Z 
2017-04-10T19:09:34.9950209Z System.AggregateException: Failed to execute the powershell script. Consult the logs below for details of the error.
2017-04-10T19:09:34.9950209Z The running command stopped because the preference variable "ErrorActionPreference" or common parameter is set to Stop: System.Management.Automation.RemoteException ---> System.Management.Automation.ActionPreferenceStopException: The running command stopped because the preference variable "ErrorActionPreference" or common parameter is set to Stop: System.Management.Automation.RemoteException
2017-04-10T19:09:34.9950209Z    at System.Management.Automation.Runspaces.PipelineBase.Invoke(IEnumerable input)
2017-04-10T19:09:34.9950209Z    at System.Management.Automation.PowerShell.Worker.ConstructPipelineAndDoWork(Runspace rs, Boolean performSyncInvoke)
2017-04-10T19:09:34.9950209Z    at System.Management.Automation.PowerShell.Worker.CreateRunspaceIfNeededAndDoWork(Runspace rsToUse, Boolean isSync)
2017-04-10T19:09:34.9950209Z    at System.Management.Automation.PowerShell.CoreInvokeHelper[TInput,TOutput](PSDataCollection`1 input, PSDataCollection`1 output, PSInvocationSettings settings)
2017-04-10T19:09:34.9950209Z    at System.Management.Automation.PowerShell.CoreInvoke[TInput,TOutput](PSDataCollection`1 input, PSDataCollection`1 output, PSInvocationSettings settings)
2017-04-10T19:09:34.9950209Z    at System.Management.Automation.PowerShell.Invoke(IEnumerable input, PSInvocationSettings settings)
2017-04-10T19:09:34.9950209Z    at Microsoft.VisualStudio.Services.DevTestLabs.RemoteDeployer.Deployment.PowershellExecutor.<>c__DisplayClass28_0.<Invoke>b__0()
2017-04-10T19:09:34.9950209Z    at Microsoft.VisualStudio.Services.DevTestLabs.RemoteDeployer.Utilities.RetryExecutor.<>c__DisplayClass19_0`1.<Execute>b__0()
2017-04-10T19:09:34.9960210Z    at Microsoft.VisualStudio.Services.DevTestLabs.RemoteDeployer.Utilities.RetryExecutor.Execute(Action action)
2017-04-10T19:09:34.9960210Z    at Microsoft.VisualStudio.Services.DevTestLabs.RemoteDeployer.Utilities.RetryExecutor.Execute[T](Func`1 action)
2017-04-10T19:09:34.9960210Z    at Microsoft.VisualStudio.Services.DevTestLabs.RemoteDeployer.Deployment.PowershellExecutor.Invoke(String errorContextMessage, Boolean writeResultToLog, Boolean isCancellable, ActionPreference errorActionPreference)
2017-04-10T19:09:34.9960210Z    --- End of inner exception stack trace ---
2017-04-10T19:09:34.9960210Z    at Microsoft.VisualStudio.Services.DevTestLabs.RemoteDeployer.Deployment.PowershellExecutor.Invoke(String errorContextMessage, Boolean writeResultToLog, Boolean isCancellable, ActionPreference errorActionPreference)
2017-04-10T19:09:34.9960210Z    at Microsoft.VisualStudio.Services.DevTestLabs.RemoteDeployer.Deployment.ScriptExecutor.ExecuteScript(IPowerShell powerShell, String script, String arguments, Boolean isScriptDotSourced, Boolean useHttp)
2017-04-10T19:09:34.9960210Z    at Microsoft.VisualStudio.Services.DevTestLabs.RemoteDeployer.Deployment.DeploymentService.RunPowerShellScripts(DeploymentMachineSpecification deploymentMachine, ScriptSpecification scriptSpecification, ScriptSpecification initializationScriptSpecification, String applicationPath, IPowerShell powerShellSession)
2017-04-10T19:09:34.9960210Z ---> (Inner Exception #0) System.Management.Automation.ActionPreferenceStopException: The running command stopped because the preference variable "ErrorActionPreference" or common parameter is set to Stop: System.Management.Automation.RemoteException
2017-04-10T19:09:34.9960210Z    at System.Management.Automation.Runspaces.PipelineBase.Invoke(IEnumerable input)
2017-04-10T19:09:34.9960210Z    at System.Management.Automation.PowerShell.Worker.ConstructPipelineAndDoWork(Runspace rs, Boolean performSyncInvoke)
2017-04-10T19:09:34.9960210Z    at System.Management.Automation.PowerShell.Worker.CreateRunspaceIfNeededAndDoWork(Runspace rsToUse, Boolean isSync)
2017-04-10T19:09:34.9960210Z    at System.Management.Automation.PowerShell.CoreInvokeHelper[TInput,TOutput](PSDataCollection`1 input, PSDataCollection`1 output, PSInvocationSettings settings)
2017-04-10T19:09:34.9960210Z    at System.Management.Automation.PowerShell.CoreInvoke[TInput,TOutput](PSDataCollection`1 input, PSDataCollection`1 output, PSInvocationSettings settings)
2017-04-10T19:09:34.9960210Z    at System.Management.Automation.PowerShell.Invoke(IEnumerable input, PSInvocationSettings settings)
2017-04-10T19:09:34.9960210Z    at Microsoft.VisualStudio.Services.DevTestLabs.RemoteDeployer.Deployment.PowershellExecutor.<>c__DisplayClass28_0.<Invoke>b__0()
2017-04-10T19:09:34.9960210Z    at Microsoft.VisualStudio.Services.DevTestLabs.RemoteDeployer.Utilities.RetryExecutor.<>c__DisplayClass19_0`1.<Execute>b__0()
2017-04-10T19:09:34.9960210Z    at Microsoft.VisualStudio.Services.DevTestLabs.RemoteDeployer.Utilities.RetryExecutor.Execute(Action action)
2017-04-10T19:09:34.9960210Z    at Microsoft.VisualStudio.Services.DevTestLabs.RemoteDeployer.Utilities.RetryExecutor.Execute[T](Func`1 action)
2017-04-10T19:09:34.9960210Z    at Microsoft.VisualStudio.Services.DevTestLabs.RemoteDeployer.Deployment.PowershellExecutor.Invoke(String errorContextMessage, Boolean writeResultToLog, Boolean isCancellable, ActionPreference errorActionPreference)<---
2017-04-10T19:09:34.9960210Z 
2017-04-10T19:09:34.9960210Z 
2017-04-10T19:09:35.0000209Z Deployment status for machine 'xyz.eastus.cloudapp.azure.com:5986' : 'Failed'
2017-04-10T19:09:35.0210207Z 
2017-04-10T19:09:35.0210207Z 
2017-04-10T19:09:35.0430206Z ##[error]System.Management.Automation.RuntimeException: The running command stopped because the preference variable "ErrorActionPreference" or common parameter is set to Stop: System.Management.Automation.RemoteException ---> System.Management.Automation.RuntimeException: The running command stopped because the preference variable "ErrorActionPreference" or common parameter is set to Stop: System.Management.Automation.RemoteException
2017-04-10T19:09:35.0430206Z    --- End of inner exception stack trace ---
2017-04-10T19:09:35.0430206Z    at System.Management.Automation.Runspaces.PipelineBase.Invoke(IEnumerable input)
2017-04-10T19:09:35.0430206Z    at System.Management.Automation.PowerShell.Worker.ConstructPipelineAndDoWork(Runspace rs, Boolean performSyncInvoke)
2017-04-10T19:09:35.0430206Z    at System.Management.Automation.PowerShell.Worker.CreateRunspaceIfNeededAndDoWork(Runspace rsToUse, Boolean isSync)
2017-04-10T19:09:35.0430206Z    at System.Management.Automation.PowerShell.CoreInvokeHelper[TInput,TOutput](PSDataCollection`1 input, PSDataCollection`1 output, PSInvocationSettings settings)
2017-04-10T19:09:35.0430206Z    at System.Management.Automation.PowerShell.CoreInvoke[TInput,TOutput](PSDataCollection`1 input, PSDataCollection`1 output, PSInvocationSettings settings)
2017-04-10T19:09:35.0430206Z    at Microsoft.TeamFoundation.DistributedTask.Handlers.LegacyVSTSPowerShellHost.VSTSPowerShellHost.Main(String[] args)
2017-04-10T19:09:35.0550190Z ##[error]LegacyVSTSPowerShellHost.exe completed with return code: -1.
2017-04-10T19:09:35.0550190Z ##[section]Finishing: Run PowerShell on xyz.eastus.cloudapp.azure.com

**Comment by pavanadepu2** [comment]
@rafaelzm2000, thank for reporting an issue.
Could you please mail the detailed log to **RM_Customer_Queries@microsoft.com**.

Note: To collect the detailed logs, set the variable "system.debug" to "true" and trigger  a release.

**Comment by arjgupta** [comment]
Hi @rafaelzm2000 , are you still facing this issue? If yes, can you please send the debug logs.

**Comment by rafaelzm2000** [comment]
Finally figured out the issue.  Because the "ErrorActionPreference" error is so broad I was not getting a lot of information to what was really causing the issue. After adding logs to the source code, I found out that what was really causing the problem was a SystemIO file not found. 
I was using PowerShell on Target Machines task to run a Powershell script which in turn would run an executable. the script was located on C:\Source on the target machine while the .exe was in C:\Windows\System32\...  In the scrip I was using absolute paths to run the executable but for some unknown reason it was not finding the executable.  it worked fine when running locally on the VM but not when using the remote task. 
The solution was to  run the script from the same directory as the executable. Once I changed the directory the task completed successfully.
